src/interpreter/index.ts

```diff
@@ -776,7 +776,61 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
     visitM(expression: Expr.M) {
         return BrsInvalid.Instance;
     }
-    visitSet(expression: Expr.Set) {
+
+    visitDottedSet(expression: Stmt.DottedSet) {
```

Comment:
Is this expression one line or can it be an entire code block?

Reply:
Whoops, forgot to rename the variable in that last commit. Should have been `statement`.

Code blocks are considered statements on their own (see `Statement.Block`. The new value from a `Set` is an expression — IIRC `Expr.Call` and `Expr.IndexedGet` are about the most complex expressions besides anonymous functions. `foo["Lorem-ipsum"] = bar.baz()` is supported. Same with anonymous functions 😉